src/main/java/org/broadinstitute/hellbender/tools/variantdb/ArrayExtractCohortEngine.java

```diff
@@ -0,0 +1,363 @@
+//            progressMeter.update(annotatedVC);
+//        }
+
+        if ( finalVC != null ) {
+            vcfWriter.add(finalVC);
+            progressMeter.update(finalVC);
+        } else {
+            // TODO should i print a warning here?
+            vcfWriter.add(mergedVC);
+            progressMeter.update(mergedVC);
+        }
+    }
+
+    private String formatFloatForVcf(final Float value) {
+        if (value == null || Double.isNaN(value)) {
+            return DOT;
+        }
+        return df.format(value);
+    }
+
+    private Float getNullableFloatFromDouble(Object d) {
+        return d == null ? null : (float)  ((Double) d).doubleValue();
+    }
+
+    private VariantContext createVariantContextFromSampleRecord(final ProbeInfo probeInfo, final GenericRecord sampleRecord, final Set<String> columnNames, final String contig, final long startPosition, final String sample) {
+        final VariantContextBuilder builder = new VariantContextBuilder();
+        final GenotypeBuilder genotypeBuilder = new GenotypeBuilder();
+
+        builder.chr(contig);
+        builder.start(startPosition);
+
+        
+        final List<Allele> alleles = new ArrayList<>();
+        Allele ref = Allele.create(probeInfo.ref, true);        
+        alleles.add(ref);
+
+        Allele alleleA = Allele.create(probeInfo.alleleA, false);
+        Allele alleleB = Allele.create(probeInfo.alleleB, false);
+
+        boolean alleleAisRef = probeInfo.ref.equals(probeInfo.alleleA);
+        boolean alleleBisRef = probeInfo.ref.equals(probeInfo.alleleB);
+
+        if (alleleAisRef) {
+            alleleA = ref;
+        } else {
+            alleles.add(alleleA);
+        }
+
+        if (alleleBisRef) {
+            alleleB = ref;
+        } else {
+            alleles.add(alleleB);
+        }
+
+        builder.alleles(alleles);
+        builder.stop(startPosition + alleles.get(0).length() - 1);
+
+        Float normx;
+        Float normy;
+        Float baf;
+        Float lrr;
+        List<Allele> genotypeAlleles = new ArrayList<Allele>();
+
+        if (this.useCompressedData) {
+            final RawArrayData data = RawArrayData.decode((Long) sampleRecord.get(SchemaUtils.RAW_ARRAY_DATA_FIELD_NAME));
+            normx = data.normx;
+            normy = data.normy;
+            lrr = data.lrr;
+            baf = data.baf;
+
+            // Genotype -- what about no-call?
+            if (data.genotype == ArrayGenotype.AA) {
+                genotypeAlleles.add(alleleA);
+                genotypeAlleles.add(alleleA);
+            } else if (data.genotype == ArrayGenotype.AB) {
+                genotypeAlleles.add(alleleA);
+                genotypeAlleles.add(alleleB);
+            } else if  (data.genotype == ArrayGenotype.BB) {
+                genotypeAlleles.add(alleleB);
+                genotypeAlleles.add(alleleB);
+            }
+        } else {
+            // TODO: constantize
+            try {
```

Comment:
i think you can use the constants from the RawArrayFieldEnum

Reply:
yeah I think these might all go away once we're storing compressed data so I didn't think too much about it